src/discord-cluster-manager/report.py

```diff
@@ -43,31 +46,51 @@ async def _send_split_log(thread: discord.Thread, partial_message: str, header:
         return ""
 
 
-async def _generate_compile_report(thread: discord.Thread, comp: CompileResult):
+@dataclasses.dataclass
+class Text:
+    text: str
+
+
+@dataclasses.dataclass
+class Log:
+    header: str
+    content: str
+
+
+class RunResultReport:
+    def __init__(self):
+        self.data: List[Text | Log] = []
+
+    def add_text(self, section: str):
+        self.data.append(Text(section))
+
+    def add_log(self, header: str, log: str):
+        self.data.append(Log(header, log))
+
+
+def _generate_compile_report(reporter: "RunResultReport", comp: CompileResult):
```

Comment:
should the CompileResult and RunResultReport also be data classed?

Reply:
`CompileResult` already is a dataclass. no strong opinion from my side for `RunResultReport`